python/ray/serve/scripts.py

```diff
@@ -465,19 +473,19 @@ def build(import_path: str, app_dir: str, output_path: Optional[str]):
         )
 
     app = build_app(node)
-
-    config = {
-        "import_path": import_path,
-        "runtime_env": {},
-        "host": "0.0.0.0",
-        "port": 8000,
-    }
-    config.update(
-        ServeApplicationSchema(
-            deployments=[deployment_to_schema(d) for d in app.deployments.values()]
-        ).dict(exclude_defaults=True)
+    schema = ServeApplicationSchema(
+        import_path=import_path,
+        runtime_env={},
+        host="0.0.0.0",
+        port=8000,
+        deployments=[deployment_to_schema(d) for d in app.deployments.values()],
     )
 
+    if kubernetes_format:
+        config = schema.kubernetes_dict(exclude_unset=True)
```

Comment:
quick non-blocking question here.. 
why do we have `kubernetes_dict` calls `dict` instead of calling dict first, then have a pure function that transform the config to k8s format? (basically why a method while the stuff is pretty pure functional)

Reply:
Good question– the Kubernetes dictionary format had a few rules that were schema-specific:

* All the keys in the config (including deployment dictionaries and `ray_actor_options`) should be camel-cased.
* Some of the dictionaries need to be json-serialized:
    * `runtime_env`
    * deployment_dictionary > `user_config`
    * deployment dictionary > `ray_actor_options` > `runtime_env`
    * deployment_dictionary > `ray_actor_options` > `resources`)

I made it a method because the second requirement made it feel too specific to be a pure function.